File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio32.java
//Uma empresa está fazendo análise de uma determinada população. Os seguintes dados são obtidos das pessoas: 
// a. Idade 
// b. Cor de Cabelos (Castanhos, Pretos, Loiros ou Outro) 
// c. Altura 
// d. Peso 
// e. Sexo (Masculino ou Feminino)
 
//Faça um algoritmo para mostrar, ao final do processamento os seguintes dados: 
// a. Quantas pessoas possuem idade superior a 18 anos 
// b. Qual é a média das idades, a idade da pessoa mais idosa e a idade da pessoa mais jovem? 
// c. Quantas pessoas em porcentagem possuem cabelos castanhos, pretos, loiros e outros? 
// d. Quantas pessoas possuem altura superior a 1,70 m 
// e. Quantas pessoas peso superior a 80 kg. 
// f. Quantas pessoas em porcentagem são do sexo masculino e quantas são do sexo feminino.
import java.util.Scanner;

public class Exercicio32 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int totalPessoas;
        int idade;
        String corCabelo;
        double altura;
        double peso;
        String sexo;

        int countIdadeSuperior18 = 0;
        int somaIdades = 0;
        int idadeMaisJovem = Integer.MAX_VALUE;
        int idadeMaisIdosa = Integer.MIN_VALUE;

        int countCabelosCastanhos = 0;
        int countCabelosPretos = 0;
        int countCabelosLoiros = 0;
        int countCabelosOutro = 0;

        int countAlturaSuperior170 = 0;
        int countPesoSuperior80 = 0;

        int countSexoMasculino = 0;
        int countSexoFeminino = 0;

        System.out.print("Digite o número total de pessoas: ");
        totalPessoas = scanner.nextInt();

        for (int i = 0; i < totalPessoas; i++) {
            System.out.println("\nDados da pessoa " + (i + 1) + ":");
            
            System.out.print("Idade: ");
            idade = scanner.nextInt();
            somaIdades += idade;

            if (idade > 18) {
                countIdadeSuperior18++;
            }

            if (idade < idadeMaisJovem) {
                idadeMaisJovem = idade;
            }

            if (idade > idadeMaisIdosa) {
                idadeMaisIdosa = idade;
            }

            System.out.print("Cor de Cabelos (Castanhos, Pretos, Loiros ou Outro): ");
            corCabelo = scanner.next().toLowerCase();

            switch (corCabelo) {
                case "castanhos":
                    countCabelosCastanhos++;
                    break;
                case "pretos":
                    countCabelosPretos++;
                    break;
                case "loiros":
                    countCabelosLoiros++;
                    break;
                default:
                    countCabelosOutro++;
                    break;
            }

            System.out.print("Altura (em metros): ");
            altura = scanner.nextDouble();
            
            if (altura > 1.70) {
                countAlturaSuperior170++;
            }

            System.out.print("Peso (em kg): ");
            peso = scanner.nextDouble();

            if (peso > 80) {
                countPesoSuperior80++;
            }

            System.out.print("Sexo (Masculino ou Feminino): ");
            sexo = scanner.next().toLowerCase();

            if (sexo.equals("masculino")) {
                countSexoMasculino++;
            } else if (sexo.equals("feminino")) {
                countSexoFeminino++;
            } else {
                System.out.println("Sexo não reconhecido.");
            }
        }

        scanner.close();

        double mediaIdades = somaIdades / (double) totalPessoas;
        double percentCabelosCastanhos = (countCabelosCastanhos / (double) totalPessoas) * 100;
        double percentCabelosPretos = (countCabelosPretos / (double) totalPessoas) * 100;
        double percentCabelosLoiros = (countCabelosLoiros / (double) totalPessoas) * 100;
        double percentCabelosOutro = (countCabelosOutro / (double) totalPessoas) * 100;
        double percentSexoMasculino = (countSexoMasculino / (double) totalPessoas) * 100;
        double percentSexoFeminino = (countSexoFeminino / (double) totalPessoas) * 100;

        System.out.println("\nResultados:");
        System.out.println("a. Pessoas com idade superior a 18 anos: " + countIdadeSuperior18);
        System.out.println("b. Média das idades: " + mediaIdades);
        System.out.println("   Idade da pessoa mais jovem: " + idadeMaisJovem);
        System.out.println("   Idade da pessoa mais idosa: " + idadeMaisIdosa);
        System.out.println("c. Porcentagem de pessoas com cabelos castanhos: " + percentCabelosCastanhos + "%");
        System.out.println("   Porcentagem de pessoas com cabelos pretos: " + percentCabelosPretos + "%");
        System.out.println("   Porcentagem de pessoas com cabelos loiros: " + percentCabelosLoiros + "%");
        System.out.println("   Porcentagem de pessoas com outros tipos de cabelo: " + percentCabelosOutro + "%");
        System.out.println("d. Pessoas com altura superior a 1.70m: " + countAlturaSuperior170);
        System.out.println("e. Pessoas com peso superior a 80 kg: " + countPesoSuperior80);
        System.out.println("f. Porcentagem de pessoas do sexo masculino: " + percentSexoMasculino + "%");
        System.out.println("   Porcentagem de pessoas do sexo feminino: " + percentSexoFeminino + "%");
    }
}

File: Livro - Módulo Python/4 - Estrutura de Repetição/Exercicio.32.py
#32 - Uma empresa está fazendo análise de uma determinada população. Os seguintes dados são obtidos das pessoas: 
# a. Idade 
# b. Cor de Cabelos (Castanhos, Pretos, Loiros ou Outro) 
# c. Altura 
# d. Peso 
# e. Sexo (Masculino ou Feminino)
# Faça um algoritmo para mostrar, ao final do processamento os seguintes dados: 
# a. Quantas pessoas possuem idade superior a 18 anos 
# b. Qual é a média das idades, a idade da pessoa mais idosa e a idade da pessoa mais jovem? 
# c. Quantas pessoas em porcentagem possuem cabelos castanhos, pretos, loiros e outros? 
# d. Quantas pessoas possuem altura superior a 1,70 m 
# e. Quantas pessoas peso superior a 80 kg. 
# f. Quantas pessoas em porcentagem são do sexo masculino e quantas são do sexo feminino. 
idade_total = 0
idade_mais_idosa = -1
idade_mais_jovem = float('inf')
idade_acima_18 = 0
total_pessoas = 0
altura_acima_170 = 0
peso_acima_80 = 0
masculino = 0
feminino = 0

while True:
    idade = int(input("Idade (-1 para encerrar): "))
    if idade == -1:
        break
    
    cor_cabelos = input("Cor de cabelos: ").lower()
    altura = float(input("Altura (m): "))
    peso = float(input("Peso (kg): "))
    sexo = input("Sexo (M/F): ").lower()
    
    idade_total += idade
    idade_mais_idosa = max(idade_mais_idosa, idade)
    idade_mais_jovem = min(idade_mais_jovem, idade)
    idade_acima_18 += idade > 18
    total_pessoas += 1
    altura_acima_170 += altura > 1.70
    peso_acima_80 += peso > 80
    masculino += sexo == 'm'
    feminino += sexo == 'f'

media_idades = idade_total / total_pessoas if total_pessoas > 0 else 0
percent_castanhos = (total_pessoas - (masculino + feminino)) / total_pessoas * 100
print(f"\nPessoas com idade acima de 18: {idade_acima_18}")
print(f"Média das idades: {media_idades:.2f}")
print(f"Pessoa mais idosa: {idade_mais_idosa}")
print(f"Pessoa mais jovem: {idade_mais_jovem}")
print(f"Porcentagem com cabelos castanhos, pretos, loiros ou outros: {percent_castanhos:.2f}%")
print(f"Pessoas com altura > 1,70m: {altura_acima_170}")
print(f"Pessoas com peso > 80kg: {peso_acima_80}")
print(f"Porcentagem de homens: {masculino / total_pessoas * 100:.2f}%")
print(f"Porcentagem de mulheres: {feminino / total_pessoas * 100:.2f}%")